go/vt/sqlparser/tracked_buffer.go

```diff
@@ -288,14 +314,26 @@ func areBothISExpr(op Expr, val Expr) bool {
 // WriteArg writes a value argument into the buffer along with
 // tracking information for future substitutions.
 func (buf *TrackedBuffer) WriteArg(prefix, arg string) {
+	length := len(prefix) + len(arg)
 	buf.bindLocations = append(buf.bindLocations, BindLocation{
```

Comment:
As a potential follow up:

I got here because `WriteArg` stood out a bit in a pprof I was looking at for heap allocations, and the other thing that can be done in here with some more work is the `bindLocations` slice that keeps getting appended to 1 at a time.

So we might want to do a thing where we count up the args first so can we grow the slice, then append to the slice. So adding a method like, `GrowArgs(size)`